wmfdata/spark.py

```diff
@@ -6,27 +6,47 @@
 
 from wmfdata.utils import check_kerberos_auth, ensure_list
 
-REGULAR_SPARK_SETTINGS = {
-    "spark.driver.memory": "2g",
-    "spark.dynamicAllocation.maxExecutors": 64,
-    "spark.executor.memory": "8g",
-    "spark.executor.cores": 4,
-    "spark.sql.shuffle.partitions": 256
+"""
+Predefined spark sessions and configs for use with the get_session and run functions.
+"""
+PREDEFINED_SPARK_SESSIONS = {
+    'local': {
+        'master': 'local[2]',  # [2] means use 2 local worker threads
+        'config': {
+            'spark.driver.memory': '4g',
+        }
+    },
+    'yarn': {
```

Comment:
There are pretty significant differences, especially regarding to dependencies.  Local has access to your whole conda env, Yarn does not.  

I guess I changed this because 'regular' sounds like it would be 'local' to me. Additionally, the size of a local process itself can be adjusted so, there might be a difference between an e.g. local-large and a yarn-large.

Would it make sense to keep the accurate terms for the predefined sessions, and add aliases like 'large' and 'small'?

Reply:
As I said in chat, this way makes sense to me now. Let's leave it as you've done it. Can you just change "yarn" to "yarn-regular" just so the progression in power is extra clear?

Reply:
Done.